src/main/java/mb/resource/DefaultResourceService.java

```diff
@@ -224,15 +232,19 @@ public HierarchicalResource appendOrReplaceWithHierarchical(HierarchicalResource
         if(registry != null) {
             return registry;
         }
-        if(parent != null) {
-            return parent.getResourceRegistry(qualifier);
-        }
-        return null;
+        return parents.stream()
+            .map(parent -> parent.getResourceRegistry(qualifier))
+            .filter(Objects::nonNull)
+            .findFirst()
+            .orElse(null);
     }
 
 
     @Override
-    public DefaultResourceService createChild(ResourceRegistry defaultRegistry, Iterable<ResourceRegistry> registries) {
-        return new DefaultResourceService(this, defaultRegistry, registries);
+    public DefaultResourceService createChild(ResourceRegistry defaultRegistry, Iterable<ResourceRegistry> registries, Iterable<ResourceService> otherParents) {
+        ArrayList<ResourceService> parents = new ArrayList<>();
+        parents.add(this);
+        otherParents.forEach(parents::add);
```

Comment:
Given how the previous implementation worked, I don't think you are supposed to add the parents of this `ResourceService` to the child. I think the parents of this `ResourceService` should be grandparents of the child, like a linked list of parents.
I don't know for sure though, I don't know the codebase very well.

Reply:
I don't know the codebase very well either, allthough I choose this implementation because:
- It reflects the mental model of the design the best
- The intended parallel change in Pie follows the same pattern.

However, we might hear Gabriels opinion about this.

Reply:
I've pushed the change.

Reply:
I don't see any changes

Reply:
Annoying. I push the changed (verified [here]https://github.com/AZWN/resource/tree/multiple-parents()), but it does not appear on this PR. I'll try to figure out.

Reply:
Repushed the commit, and now it shows up.